vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -32,23 +34,19 @@ def save_inventory_prep(find_key, child_keys, extra_keys)
     return find_key, child_keys, extra_keys, remove_keys
   end
 
-  def save_inventory(type, klass, parent, hash, deletes, new_records, record_index, record_index_columns, find_key, child_keys, remove_keys)
-    # Backup keys that cannot be written directly to the database
-    key_backup = backup_keys(hash, remove_keys)
-
+  def save_inventory(type, parent, hash, deletes = nil, new_records = nil, record_index = nil,
+                     record_index_columns = nil, find_key = nil)
     # Find the record, and update if found, else create it
     found = find_key.blank? ? parent.send(type) : self.save_inventory_record_index_fetch(record_index, record_index_columns, hash, find_key)
     if found.nil?
-      found = klass.new(hash)
-      new_records.nil? ? parent.send("#{type}=", found) : new_records << found
+      found = find_key ? parent.send(type).build(hash) : parent.send("build_#{type}", hash)
+      # TODO: these should not be necessary. Probably missing some parent.save calls
```

Comment:
Rails doesn't batch insert the "right" way, but it would be sweet if it did.  It was on my long list of "I'm gonna add this to Rails one day".

I wrote the code to expect that in the future, while still leveraging what it does as "batch" now, which is to do it all in one transaction.  With, say, 1000s of inserts, that's a lot of BEGIN/COMMIT pairs that are avoided even without the single insert statement.